tests/rpmi.at

```diff
@@ -1371,3 +1371,21 @@ runroot rpm -q --whatprovides /
 ],
 [])
 AT_CLEANUP
+
+AT_SETUP([rpm -U with Obsoletes])
+AT_KEYWORDS([install])
+AT_CHECK([
+RPMDB_INIT
+
+runroot rpmbuild -bb --quiet /data/SPECS/test-obsoletes.spec
+runroot rpm -i --nodeps --ignorearch /data/RPMS/hello-1.0-1.i386.rpm
+runroot rpm -U /build/RPMS/noarch/test-obsoletes-1.0-1.noarch.rpm
```

Comment:
Heh, good spotting that we seem to have no test for the plain vanilla package-obsoletes-another case.

But do use the existing test material, you can just build another variant at the start of the "rpm -U obsoleted packages" test and the test against it in its own AT_CHECK group just like the other obsoletes-tests. That also avoids the problematic hello package which should only be used for special cases.

Reply:
I'm sorry I didn't quite understand you, but I made some changes. Please check if it's OK.

Reply:
I meant: use the existing deptest.spec to create the test-material you need. This can (and should) go into the existing obsoletes test group (see https://github.com/rpm-software-management/rpm/blob/db1c9da7fe507b7eb0424bf9c76b6cf37094ed36/tests/rpmi.at#L988), and because there's already test-material being built, you only need to build one additional package for this test.

Reply:
Thank you very much for your tip, I have made relevant changes. Please review the changes.

Reply:
I don't see any changes made to this part, it's still adding the new spec unnecessarily.

Reply:
Oh, I understand what you mean, and it's now been revised.